retrievalmarket/impl/clientstates/client_fsm.go

```diff
@@ -105,8 +110,18 @@ var ClientEvents = fsm.Events{
 			return nil
 		}),
 
+	// The payment channel add funds message has landed on chain
 	fsm.Event(rm.ClientEventPaymentChannelReady).
+		// If the payment channel between client and provider was being created
+		// for the first time, or if the payment channel had already been
+		// created for an earlier deal but the initial funding for this deal
+		// was being added, then we still need to allocate a payment channel
+		// lane
 		From(rm.DealStatusPaymentChannelCreating).To(rm.DealStatusPaymentChannelAllocatingLane).
+		From(rm.DealStatusPaymentChannelAddingInitialFunds).To(rm.DealStatusPaymentChannelAllocatingLane).
```

Comment:
I am a bit confused with naming here. Why `Initial` funds? If the payment channel existed and ran out of funds, wouldn't it be better to call it `AddingMoreFunds`?

Reply:
I guess it is `Initial`, because this is adding funds for a new deal, right?

Reply:
Right exactly, the funds are for the new deal 👍